### DISCUSSION TITLE
Unstable receiving of emits from server

### DISCUSSION THREAD

**Discussion by marmagzzz** [socketio/socket.io#3774] [Q&A] [answered]
I have this chat application using socketio for the realtime feature (messages, notifications, etc.).

When it's in my local, the receiving of emits from client to server (vice versa) is working fine.
Messages and notifications are being received immediately.
But when I deployed it to my live server, there are times that emit from server **doesn't** received by the client.

Config for server:

`const socketIo = require('socket.io')(server, { transports: ['websocket'], wsEngine: 'ws' });`

Config for client:

`const socketIoConnection = io(ROOT_URL, { transports : ['websocket'], secure : true });`

I also use pm2 for node clustering on server

I've been stuck here for almost 2 weeks from now.
Any answer, idea or help will be highly appreciated.
Thank you!

**Comment by marmagzzz** [comment, marked as answer]
I tried my server to disable cluster mode of pm2 and it worked fine as expected.
I will just follow this to make it work on cluster mode of pm2.
https://github.com/elad/node-cluster-socket.io

**Reply by darrachequesne** [reply]
Please note that, if you are only the `websocket` transport, you don't need sticky-session (either based on the user IP address, in https://github.com/elad/node-cluster-socket.io, or on the session ID in https://github.com/darrachequesne/socket.io-sticky).

In order to broadcast packets (`io.emit()` or `socket.to().emit()` for example), the Redis adapter (or any compatible adapter) is needed though.

Documentation: https://socket.io/docs/v3/using-multiple-nodes/
